services/gemma-local/main.py

```
@@ -806,23 +953,28 @@
         except Exception as e:
             logger.error(f"Generation failed on {model_manager.device}: {e}")
             # Fallback to simpler generation with configured limits
+            if model_manager.model is None or model_manager.tokenizer is None:
+                raise ValueError("Model or tokenizer not loaded")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)